src/kudu/consensus/consensus_queue.cc

```diff
@@ -1266,13 +1361,23 @@ bool PeerMessageQueue::ResponseFromPeer(const std::string& peer_uuid,
     // the same watermarks and make the same advancement, so this is safe.
     if (mode_copy == LEADER) {
       // Advance the majority replicated index.
-      AdvanceQueueWatermark("majority_replicated",
-                            &queue_state_.majority_replicated_index,
-                            /*replicated_before=*/ prev_peer_state.last_received,
-                            /*replicated_after=*/ peer->last_received,
-                            /*num_peers_required=*/ queue_state_.majority_size_,
-                            VOTER_REPLICAS,
-                            peer);
+      if (!FLAGS_enable_flexi_raft) {
+        AdvanceQueueWatermark("majority_replicated",
+                              &queue_state_.majority_replicated_index,
+                              /*replicated_before=*/ prev_peer_state.last_received,
+                              /*replicated_after=*/ peer->last_received,
+                              /*num_peers_required=*/ queue_state_.majority_size_,
+                              VOTER_REPLICAS,
+                              peer);
+      } else {
+        AdvanceMajorityReplicatedWatermark(
+            &queue_state_.majority_replicated_index,
+            /*replicated_before=*/ prev_peer_state.last_received,
+            /*replicated_after=*/ peer->last_received,
+            /*data_commit_quorum=*/ GetDataCommitQuorum(),
```

Comment:
cmeta_ is a memeber of PeerMessageQueue and AdvanceMajorityReplicatedWatermark() has access to cmeta_ and can compute data commit quorum itself.